src/Settings/Builders/Ghc.hs

```diff
@@ -28,7 +29,9 @@ ghcLinkArgs = builder (Ghc LinkHs) ? do
                    buildInfo <- lift $ readFileLines gmpBuildInfoPath
                    return $ concatMap (words . strip) buildInfo
                else return []
-    mconcat [ arg "-no-auto-link-packages"
+    mconcat [ (Dynamic `wayUnit` way) ?
+              append [ "-shared", "-dynamic", "-dynload", "deploy" ]
+            , arg "-no-auto-link-packages"
```

Comment:
Isn't it still on the line after the `append`? Both static and dynamic needs that flag I think.